### PR TITLE
Clean up hs

### PR DESCRIPTION
**Pull request by Ian-Stewart-Binks**
This pull request cleans up a lot of the Haskell code.

#305 must be merged/rejected before this PR is considered.

### CODE REVIEW

File: hs/GraphResponse.hs [hunk 1]
```diff
@@ -29,13 +28,13 @@ graphResponse =
 
 fceCountDiv :: H.Html
 fceCountDiv = createTag H.div "FCECountDiv" "" $ do
-                  createTag H.span "FCEcount" "" $ do "0.0"
+                  createTag H.span "FCEcount" "" "0.0"
                   "FCEs" -- Being difficult. Won't show up correctly.
 
 welcomeTab :: H.Html
-welcomeTab = createTag H.div "welcome" "infoTab" $ do
```
Comment (david-yz-liu) [inline hs/GraphResponse.hs]: You can go ahead and delete this whole tab.

File: hs/GraphResponse.hs [hunk 2]
```diff
@@ -70,16 +69,13 @@ focusesTab = ""
...
-                                   makeA "" "" "timetable_creator.html" "" $ do
-                                       "Plan your timetable here!"
+                               createTag H.p "timetable-creator-link" "" $
+                                   makeA "" "" "timetable_creator.html" "" "Plan your timetable here!"
                                makeInput "filter" "text-input" "" "" "text"
-                           createTag H.div "timetableContainer" "" $ do
-                               ""
+                           createTag H.div "timetableContainer" "" ""
 
 checkMyPostTab :: H.Html
```
Comment (david-yz-liu) [inline hs/GraphResponse.hs]: This one too.

File: hs/MakeElements.hs [hunk 3]
```diff
@@ -7,10 +7,10 @@ import qualified Text.Blaze.Html5.Attributes as A
 import Control.Monad    (msum)
 
 insertSVG :: H.AttributeValue -> H.Html
-insertSVG src = H.object ! A.data_ src ! A.type_ "image/svg+xml" $ do ""
+insertSVG src = H.object ! A.data_ src ! A.type_ "image/svg+xml" $ ""
```
Comment (david-yz-liu) [inline hs/MakeElements.hs]: You don't need the final `$` either

File: hs/database/CourseQueries.hs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        return $ toResponse $ createJSONResponse $ encodeJSON $ Aeson.toJSON courseJSON
+
+-- | Builds a Course structure from a tuple from the Courses table.
+-- Some fields still need to be added in.
+buildCourse :: Maybe Session -> Maybe Session -> Maybe Session -> Courses -> Course 
+buildCourse fallSession springSession yearSession course = 
+    Course (coursesBreadth course)
+           (coursesDescription course)
+           (coursesTitle course)
+            Nothing --prereqString
```
Comment (david-yz-liu) [inline hs/database/CourseQueries.hs]: The alignment here is off

File: hs/database/JsonParser.hs [hunk 5]
```diff
@@ -34,21 +28,21 @@ courseDirectory = "../../res/courses/"
 
 -- | Opens a directory contained in dir, and processes every file in that directory.
 processDirectory :: IO ()
-processDirectory = getDirectoryContents courseDirectory >>= \contents ->
-                    let formattedContents = map (courseDirectory ++) (L.sort contents)
-                    in filterM doesFileExist formattedContents >>= mapM_ printFile
+processDirectory = 
+    getDirectoryContents courseDirectory >>= \contents ->
```
Comment (david-yz-liu) [inline hs/database/JsonParser.hs]: Switch this back to do notation

File: hs/MakeElements.hs (hunk 3 above)
Comment (Ian-Stewart-Binks) [inline hs/MakeElements.hs]: Apparently it's necessary.

File: hs/GridResponse.hs [hunk 6]
```diff
@@ -33,22 +33,22 @@ conflictDialog :: H.Html
 conflictDialog = createTag H.div "dialog" "" "Conflicting courses are difficult to manage. Make sure you understand the added responsibility of having two or more conflicting courses."
 
 coursePanel :: H.Html
-coursePanel = createTag H.div "course-select-wrapper" "col-md-2 col-xs-6" $ do
-                            createTag H.ul "course-select" "trapScroll-enabled" $ do
-                                   createTag H.li "clear-all" "" $ do
+coursePanel = createTag H.div "course-select-wrapper" "col-md-2 col-xs-6" $
+                            createTag H.ul "course-select" "trapScroll-enabled" $
```
Comment (david-yz-liu) [inline hs/GridResponse.hs]: The indentation looks like it's using tabs in this file.

File: hs/MakeElements.hs (hunk 3 above)
Comment (david-yz-liu) [inline hs/MakeElements.hs]: Ah, yes you're right! My bad.

File: hs/MakeElements.hs [hunk 7]
```diff
@@ -53,4 +53,4 @@ aboutLinks = concatHtml (map stylesheet ["//netdna.bootstrapcdn.com/bootstrap/3.
                                        "static/style/common/common.css"])
 
 concatHtml :: [H.Html] -> H.Html
-concatHtml html = foldl (>>) "" $ html
\ No newline at end of file
+concatHtml html = foldl (>>) "" html
```
Comment (david-yz-liu) [inline hs/MakeElements.hs]: We since discovered a better way of doing this using `mapM_`, I believe.

File: hs/database/JsonParser.hs [hunk 8]
```diff
@@ -57,85 +51,93 @@ getJSON jsonFile = B.readFile jsonFile
...
+    insertSessionTutorials (f course) "F" course >>
+    insertSessionTutorials (s course) "S" course >>
+    insertSessionTutorials (y course) "Y" course
 
 -- | Inserts the tutorials from a specified section into the Tutorials table.
 insertSessionTutorials :: Maybe Session -> T.Text -> Course -> IO ()
 insertSessionTutorials Nothing sessionStr course = return ()
-insertSessionTutorials (Just session) sessionStr course = when (not $ null (tutorials session)) $ liftIO $ mapM_ ((insertTutorial sessionStr) course) (tutorials session)
+insertSessionTutorials (Just session) sessionStr course = 
+    (unless $ null (tutorials session)) $
```
Comment (david-yz-liu) [inline hs/database/JsonParser.hs]: Are the parentheses around `unless` necessary?

File: hs/server.hs [hunk 9]
```diff
@@ -44,90 +19,9 @@ main = do
     let staticDir = encodeString $ parent $ decodeString cwd
     contents <- readFile "../README.md"
     simpleHTTP nullConf $
-      msum [ dir grid $ gridResponse,
-             dir graph $ graphResponse,
-             dir about $ aboutResponse contents,
-             dir static $ serveDirectory EnableBrowsing [] staticDir,
-             dir course $ path (\s -> liftIO $ queryCourse s)
+      msum [ dir "grid" $ gridResponse,
```
Comment (david-yz-liu) [inline hs/server.hs]: Remove the `$`

File: hs/GridResponse.hs (hunk 6 above)
Comment (Ian-Stewart-Binks) [inline hs/GridResponse.hs]: Sure does look that way, but it's spaces! I'll do something about the indentation, though.

File: hs/MakeElements.hs (hunk 7 above)
Comment (Ian-Stewart-Binks) [inline hs/MakeElements.hs]: Do you mean `sequence_`?

File: hs/database/CourseQueries.hs (hunk 4 above)
Comment (Ian-Stewart-Binks) [inline hs/database/CourseQueries.hs]: I do it on purpose. I'll switch it back.

Comment (Ian-Stewart-Binks) [inline hs/database/CourseQueries.hs]: Done.

File: hs/database/JsonParser.hs (hunk 5 above)
Comment (Ian-Stewart-Binks) [inline hs/database/JsonParser.hs]: Done. Much nicer.

File: hs/database/JsonParser.hs (hunk 8 above)
Comment (Ian-Stewart-Binks) [inline hs/database/JsonParser.hs]: Yes.

File: hs/server.hs (hunk 9 above)
Comment (Ian-Stewart-Binks) [inline hs/server.hs]: Done. I tried this before, but for some reason it didn't work. Worked this time. Hmm.

**Comment by david-yz-liu** [comment]
Thanks, Ian!

**Pull request merged**